src/errors.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * Base class for custom errors in Rhum.
+ */
+class RhumError extends Error {
+  public name: string;
+
+  constructor(name: string, message: string) {
+    super(message);
+    this.name = name;
+  }
+}
+
+/**
+ * Error to throw in relation to fake logic.
+ */
+export class FakeError extends RhumError {
+  constructor(message: string) {
+    super("FakeError", message);
+  }
+}
+
+/**
+ * Error to throw in relation to method verification logic. For example, when a
+ * method is being verified that it was called once via
+ * `mock.method("doSomething").toBeCalled(1)`.
+ */
+export class MethodVerificationError extends RhumError {
+  #actual_results: string;
+  #code_that_threw: string;
+  #expected_results: string;
+
+  /**
+   * @param message - The error message.
+   */
+  constructor(
+    message: string,
+    codeThatThrew: string,
+    actualResults: string,
+    expectedResults: string,
+  ) {
+    super("MethodVerificationError", message);
+    this.#code_that_threw = codeThatThrew;
+    this.#actual_results = actualResults;
+    this.#expected_results = expectedResults;
+    this.#makeStackConcise();
+  }
+
+  /**
+   * Shorten the stack trace to show exactly what file threw the error. For
+   * example:
+   *
+   * ```ts
+   *
+   * ```
+   */
+  #makeStackConcise(): void {
+    const ignoredLines = [
+      "<anonymous>",
+      "deno:runtime",
+      "method_verifier.ts",
+      "_mixin.ts",
+    ];
+
+    if (this.stack) {
```

Comment:
possible to return early?

Reply:
true. early return is better: [4785202](https://github.com/drashland/rhum/pull/163/commits/47852029b4175d7213dc2a44a049025180656940)